torch/fx/experimental/proxy_tensor.py

```diff
@@ -47,8 +47,13 @@
 CONSTANT_NUMEL_LIMIT = 1
 
 # We currently convert all SymInt to proxies before we use them.
-# This could plausibly be handled at the Dynamo level.
-pytree._register_pytree_node(torch.Size, lambda x: (list(x), None), lambda xs, _: tuple(xs))
+# Hence we transform any torch.Size node into an equivalent tuple
+# if any of its items are not int/SymInt
+pytree._register_pytree_node(
```

Comment:
Could we move this into `torch/fx/_pytree.py` too? :thinking:

Reply:
It's not a good UX for things to be a pytree sometimes when certain submodules of torch are imported. IMO all of our pytree registrations should go into torch/utils/_pytree

Reply:
true. It would mean either:
1. We support `torch.Size` as an official pytree node
2. We restrict `torch.Size` as a node to only be a pytree node under a context manager, and run a `with pytree._register_node(*torch_size_registration)`

Reply:
since vmap also relies on it though, where will the vmap code eventually live?

Reply:
I think we will move https://github.com/pytorch/pytorch/blob/740d636165676d4a7286f05043e47db86efdeee1/torch/onnx/_internal/fx/dynamo_graph_extractor.py#L31
into utils.pytree, and then we provide a default `torch.Size` extension.

```python
from utils.pytree import register_torch_size

@register_torch_size
def my_function_containing_pytree_that_must_walk_torch_size():
  ...
```